src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/utils/SPARQL2GraphQLHelper.java

```diff
@@ -0,0 +1,249 @@
+     */
+    public static void addScalarField(final Set<String> fieldPaths, final GraphQL2RDFConfiguration config, 
+            final String currentPath, final String predicateURI){
+        final String alias = config.removePropertyPrefix(predicateURI);
+        final String fieldName = config.removePropertySuffix(alias);
+        fieldPaths.add(GraphQLFieldPathBuilder.addScalar(currentPath, alias, fieldName));
+    }
+
+
+
+    /**
+     * Helper function which checks if any triple pattern predicate in @param sgp is a variable or blank node, returns true if so
+     */
+    public static boolean hasVariablePredicate(final Set<TP> sgp){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            if(predicate.isVariable()){
+                return true;
+            }    
+        }
+        return false;
+    }
+
+    /**
+     * Returns the GraphQL object type @param sgp correponds to. 
+     * If the type is undeterminable @return null
+     */
+    public static String determineSgpType(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                return config.getClassFromPropertyURI(predicate.getURI());
+            }
+            else if(predicate.isURI() && predicate.getURI().equals(config.getClassMembershipURI())){
+                if(object.isURI() && object.getURI().startsWith(config.getClassPrefix())){
+                    final int splitIndex = config.getClassPrefix().length();
+                    return object.getURI().substring(splitIndex);
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * @return a map consisting of what can be used as arguments from the given @param sgp
+     * The predicate from a TP needs to be a property URI and the object needs to be a literal
+     */
+    public static final Map<String,LiteralLabel> getSgpArguments(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        final Map<String,LiteralLabel> args = new HashMap<>();
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                if(object.isLiteral() && object.getLiteral().isWellFormed()){
+                    // Remove prefix and suffix before adding to map
+                    final String s = config.removePropertyPrefix(predicate.toString());
+                    final String argName = config.removePropertySuffix(s);
```

Comment:
this mapping should be handled by the configuration (same with the predicate condition in line 185)